test/unit/range/iterator_test_template.hpp

```diff
@@ -88,43 +88,28 @@ TYPED_TEST_P(iterator_fixture, concept_check)
                   "the returned iterator must model std::input_iterator.");
     EXPECT_TRUE(std::input_iterator<iterator_type>);
 
-    if constexpr (std::derived_from<typename TestFixture::iterator_tag, std::forward_iterator_tag>)
-    {
-        static_assert(std::forward_iterator<decltype(std::ranges::begin(this->expected_range))>,
-                      "expected_range must have a begin member function and "
-                      "the returned iterator must model std::forward_iterator.");
-        EXPECT_TRUE(std::forward_iterator<iterator_type>);
+    EXPECT_EQ(std::forward_iterator<iterator_type>,
+              (std::derived_from<typename TestFixture::iterator_tag, std::forward_iterator_tag>));
```

Comment:
I removed those static asserts, because they get in your way when you are trying to implement an iterator. They just error and give you not a good explanation why something didn't work. For example, I had forgotten somewhere a const on a member function within the iterator and therefore `begin` wasn't returning a valid iterator. The concept couldn't tell me why my "range" wasn't a range, it just said that my begin call was invalid. Our later functional test made it clear which function I forgot to add const to, but it was buried in a see of static asserts which I had to manually comment out.

A static assert is great for giving you a full-stop signal that indicates that something does not work (anymore), but this is only useful for existing code.

We could also move these calls to the bottom of the file s.t. they don't give a warning/error first.

At the bottom-line, this code wouldn't be merged if we a) have a build-error or b) if the tests don't pass.

Reply:
I don't care, I just didn't want to overrule @rrahn's vote on this because I am second reviewer now. You can decide whether to involve him again or not, it's your call, I was just trying to be attentive to his opinion :)

Reply:
I pinged him, let's see what he thinks. :)

Reply:
Well, if the diagnostics improve by having a run time check, then I am in favor of the run time test rather the compile error. Since it is a test and the test fails this is ok. Otherwise, I prefer the earliest possible time to emit an error.